2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+# Manage Workflow for Instructor Training
+
+## Abstract
+
+Software Carpentry has been training scientists with computing skills since the last 18 years, and the number of instructors has been growing rapidly since its inception. More and more people are applying for instructor training programme which makes it difficult to manage the whole process manually. To help ease the problem, this project helps admins manage the whole workflow of instructor training starting from application to final checkout. This would help orgainizing the currently scattered workflow, getting rid of scattered emails and what not.
+
+## Proposed Workflow
+
+The whole system to manage instructor training is supposed to have two interfaces:
+
+1. The Trainee Admin - An administrative interface - which is used by admins to manage trainees at each stage, and
+
+2. The Trainee Dashboard - the one used by trainees to keep track of their progress.
+
+The process starts with applicants applying for instructor training programmes
+
+### 1. The Application Form
+
+The application process for becoming an instructor starts here. People aspiring to become instructors shall start by filling up this application form.
+
+Trainees apply in one of the following ways:
+
+  1. **Apply individually**:
+
+  This would be a normal application, requesting the relevant details from the applicant.
+
+  **Helping individuals form groups**
+
+   * Apart from the regular details, the applicant would also be asked if they are willing to pair up.
+   * If yes, they would be presented with a list of individuals or groups willing to accept new members. The applicant can then request to pair up with the individual/group of their choice.
```

Comment:
Yes, *if* we're willing to form groups. Suppose I say I want instructor training, but I don't want other people in my lab to know (because I'm supposed to be focusing on my thesis). How do we handle that?